maven-extension/src/main/java/io/opentelemetry/maven/OpenTelemetrySdkService.java

```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package io.opentelemetry.maven;
+
+import io.opentelemetry.api.GlobalOpenTelemetry;
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.common.AttributesBuilder;
+import io.opentelemetry.api.trace.Tracer;
+import io.opentelemetry.api.trace.propagation.W3CTraceContextPropagator;
+import io.opentelemetry.context.propagation.ContextPropagators;
+import io.opentelemetry.exporter.otlp.trace.OtlpGrpcSpanExporter;
+import io.opentelemetry.exporter.otlp.trace.OtlpGrpcSpanExporterBuilder;
+import io.opentelemetry.maven.semconv.MavenOtelSemanticAttributes;
+import io.opentelemetry.sdk.OpenTelemetrySdk;
+import io.opentelemetry.sdk.common.CompletableResultCode;
+import io.opentelemetry.sdk.resources.Resource;
+import io.opentelemetry.sdk.trace.SdkTracerProvider;
+import io.opentelemetry.sdk.trace.export.BatchSpanProcessor;
+import io.opentelemetry.sdk.trace.export.SpanExporter;
+import io.opentelemetry.semconv.resource.attributes.ResourceAttributes;
+import java.time.Duration;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import org.apache.maven.rtinfo.RuntimeInformation;
+import org.codehaus.plexus.component.annotations.Component;
+import org.codehaus.plexus.component.annotations.Requirement;
+import org.codehaus.plexus.personality.plexus.lifecycle.phase.Disposable;
+import org.codehaus.plexus.personality.plexus.lifecycle.phase.Initializable;
+import org.codehaus.plexus.personality.plexus.lifecycle.phase.InitializationException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Service to configure the {@link OpenTelemetry} instance.
+ *
+ * <p>Mimic the <a
+ * href="https://github.com/open-telemetry/opentelemetry-java/tree/main/sdk-extensions/autoconfigure">OpenTelemetry
+ * SDK Autoconfigure</a> that can't be used due to class loading issues when declaring the Maven
+ * OpenTelemetry extension using the pom.xml {@code <extension>} declaration: the OpenTelemetry SDK
+ * Autoconfigure registers a JVM shutdown hook that is incompatible with the fact that Maven
+ * extensions are unloaded before the JVM shuts down, causing {@link NoClassDefFoundError} when the
+ * shutdown hook is invoked.
+ */
+@Component(role = OpenTelemetrySdkService.class, hint = "opentelemetry-service")
+public final class OpenTelemetrySdkService implements Initializable, Disposable {
+
+  private final Logger logger = LoggerFactory.getLogger(getClass());
```

Comment:
```suggestion
  private static final Logger logger = LoggerFactory.getLogger(OpenTelemetrySdkService.class);
```

We always use static for loggers, and hence we never use getClass